crates/ef-testing/src/storage/mod.rs

```diff
@@ -78,12 +76,15 @@ pub fn write_test_state(
         let address =
             StarknetContractAddress(Into::<StarkFelt>::into(starknet_address).try_into()?);
         let account_nonce: FieldElement = Felt252Wrapper::try_from(account_info.nonce.0)?.into();
-        let storage_record = StorageRecord {
-            nonce: Nonce(StarkFelt::from(account_nonce)),
-            class_hash: ClassHash(class_hashes.proxy_class_hash.into()),
-            storage: starknet_contract_storage.into_iter().collect(),
-        };
-        starknet.storage.insert(address, storage_record);
+        starknet
+            .set_class_hash_at(address, ClassHash(class_hashes.proxy_class_hash.into()))
+            .map_err(|err| {
+                RunnerError::Other(format!("error setting class hash at {address:#?}: {err}"))
```

Comment:
there are three times RunnerError::Other that are linked to Katana context, should we create a RunnerError::SequencerError or some similar error to express a failure with underlying engine?